src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -209,13 +203,23 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     request.Headers.TransferEncodingChunked = true;
                 }
 
+                if (request.Version.Major == 1 && request.Version.Minor == 0 &&
+                        request.HasHeaders && request.Headers.TransferEncodingChunked == true)
+                {
+                    // HTTP 1.0 does not support chunking
+                    throw new NotSupportedException(SR.net_http_unsupported_chunking);
+                }
+
                 // Write request line
                 await WriteStringAsync(request.Method.Method, cancellationToken).ConfigureAwait(false);
                 await WriteByteAsync((byte)' ', cancellationToken).ConfigureAwait(false);
                 await WriteStringAsync(
                     _usingProxy ? request.RequestUri.AbsoluteUri : request.RequestUri.PathAndQuery,
                     cancellationToken).ConfigureAwait(false);
-                await WriteBytesAsync(s_spaceHttp11NewlineAsciiBytes, cancellationToken).ConfigureAwait(false);
+
+                // fall-back to 1.1 for all versions other than 1.0
+                await WriteBytesAsync( request.Version.Major == 1 && request.Version.Minor == 0 ?
```

Comment:
Since we're doing the 1.0 check more than once, should we maybe just save it into a single bool (or enum)?

Reply:
@wfurt, I assume we're waiting for these comments to be addressed before merging this?

Reply:
should I add the bool flag as @bartonjs suggested @stephentoub ?
I walk through the list and this seems like last issue aside from WS.

Reply:
Well... maybe.  Normally I'd say "yes, store it in a local so it doesn't need to be recomputed."  However, this is an async method, and because that local would be written before and read after an await, it'll need to be lifted to the state machine.  I've not looked at the state machine recently to see what fields are on it... it's quite likely that adding a bool to it won't actually increase its size due to it getting rounded up to the nearest word size, but it'd be worth double-checking that's the case.  Why don't you add it for now, and we can review all of the state machines later to evaluate.